content/developer/tutorials/data_modules.rst

```diff
@@ -0,0 +1,906 @@
+==================
+Write Data Modules
+==================
+
+.. important::
+   This tutorial assumes familiarity with the :doc:`server_framework_101` tutorial and the
+   :doc:`define_module_data` tutorial.
+
+Although as developpers, we prefer to have the full power of Python to write our modules,
+it is sometimes not possible to do so; typically on managed hosting solutions which do not
+allow the deployment of custom Python code like the `Odoo.com <https://www.odoo.com/start>`_
+platform.
+
+However, the flexible nature of Odoo is meant to allow customizations out of the box. Whilst
+a lot is possible with :doc:`Studio </applications/studio>`, it is also possible to define
+models, fields and logic in :doc:`XML Data File <define_module_data>`. This makes it easier
+to develop, maintain and deploy these customizations.
+
+In this tutorial, we will learn how to define models, fields and logic in XML data files.
+We will also see the limitations of this approach to module development.
+
+Problem Statement
+=================
+
+Like in the :doc:`server_framework_101` tutorial, we will be working on Real Estate concepts.
+
+Our goal is to create a new application to manage Real Estate properties in a similar (albeit
+simpler) way to the :doc:`server_framework_101` tutorial. We will define the models, fields and
+logic in XML data files instead of Python files.
+
+At the end of this tutorial, we will be able to achieve the following in our app:
+
+- Manage Real Estate properties that are for sale
+- Publish these properties on a website
+- Accept offers online from the website
+- Invoice the buyer when the property is sold
+
+Module Structure
+================
+
+Like in any development project, a clear structure makes it easier to manage and maintain the code.
+
+Unlike standard Odoo modules that use both Python and XML files, data modules use only XML files.
+Therefore, it is expected that your work tree will look something like this:
+
+.. code-block:: bash
+
+  estate
+  ├── actions
+  │   └── *.xml
+  ├── models
+  │   └── *.xml
+  ├── security
+  │   └── ir.model.access.csv
+  ├── views
+  │   └── *.xml
+  ├── __init__.py
+  └── __manifest__.py
+
+The only Python files you will have are the ``__init__.py`` and ``__manifest__.py`` files. The
+``__manifest__.py`` file will be the same as for any Odoo module, but will also import its models
+in the `data` list.
+
+Remember to list files in the ``data`` section of ``__manifest__.py`` in order of dependency,
+typically starting with model files.
+
+The ``__init__.py`` file is empty, but is required for Odoo to recognize the module.
```

Comment:
huh, did not know that!

but if you don't include it, it means your module can only ever be deployed as a data module (e.g. you can't put it in an addons-path and install it 'normally' because odoo does see it as a module)

Reply:
added some clarification